File: problem/baekjoon/python/10773.py
def zero():
    _k = int(input())

    _stack = []
    _answer = 0

    for i in range(_k):
        _input = int(input())

        if _input == 0:
            _stack.pop()
        else:
            _stack.append(_input)

    for i in _stack:
        _answer += i

    print(_answer)


zero()

File: problem/baekjoon/nodejs/10773.js
// 제로
const readline = require("readline");

const rl = readline.createInterface({
  input: process.stdin,
  output: process.stdout,
});

let input = [];

rl.on("line", function (line) {
  input.push(line);

  K = Number(input[0]);

  if (input.length - 1 === K) {
    rl.close();
  }
}).on("close", function () {
  let answer = [];

  Ks = input.slice(1).map((el) => {
    return Number(el);
  });

  for (let i = 0; i < Ks.length; i++) {
    if (Ks[i] !== 0) {
      answer.push(Ks[i]);
    } else {
      answer.pop(answer[answer.length - 1]);
    }
  }

  if (answer.length === 0) {
    console.log(0);
  } else {
    console.log(
      answer.reduce((acc, cur) => {
        return acc + cur;
      })
    );
  }

  process.exit();
});
